<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from robot_core.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot>
  <!-- Specify some standard inertial calculations https://en.wikipedia.org/wiki/List_of_moments_of_inertia -->
  <!-- These make use of xacro's mathematical functionality -->
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <material name="orange">
    <color rgba="1 0.3 0.1 1"/>
  </material>
  <material name="blue">
    <color rgba="0.2 0.2 1 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="red">
    <color rgba="1 0 0 1"/>
  </material>
  <material name="grey">
    <color rgba="0.5 0.5 0.5 1"/>
  </material>
  <!-- Create fixed world frame -->
  <link name="world"/>
  <joint name="world_joint" type="floating">
    <parent link="world"/>
    <child link="chassis"/>
    <origin xyz="0 0 0.1"/>
  </joint>
  <link name="chassis">
    <visual>
      <geometry>
        <box size="0.3 0.015 0.015"/>
      </geometry>
      <material name="orange"/>
      <origin xyz="0 -0.0575 0"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.3 0.015 0.015"/>
      </geometry>
      <origin xyz="0 0.2575 0"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.015 0.3 0.015"/>
      </geometry>
      <material name="grey"/>
      <origin xyz="-0.15 0 0"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.015 0.3 0.015"/>
      </geometry>
      <material name="grey"/>
      <origin xyz="0.15 0 0"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.3 0.015 0.015"/>
      </geometry>
      <origin xyz="0 0 0"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.015 0.3 0.015"/>
      </geometry>
      <origin xyz="0 0.2 0"/>
    </visual>
    <inertial>
      <mass value="1"/>
      <inertia ixx="1e-3" ixy="0.0" ixz="0.0" iyy="1e-3" iyz="0.0" izz="1e-3"/>
    </inertial>
  </link>
  <gazebo reference="chassis">
    <material>Gazebo/White</material>
  </gazebo>
  <!-- Left_wheel_base-->
  <link name="left_wheel_base">
    <visual>
      <geometry>
        <box size="0.015 0.015 0.05"/>
      </geometry>
      <origin xyz="-0.08 0 -0.0325"/>
      <material name="white"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.017 0.015 0.015"/>
      </geometry>
      <origin xyz="-0.08 0 -0.0575"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.015 0.015 0.076"/>
      </geometry>
      <origin xyz="-0.10450000000000001 -0.038 0.0"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.0115 0.015 0.015"/>
      </geometry>
      <origin xyz="-0.00575 0 -0.14100000000000001"/>
    </visual>
    <inertial>
      <mass value="1"/>
      <inertia ixx="1e-3" ixy="0.0" ixz="0.0" iyy="1e-3" iyz="0.0" izz="1e-3"/>
    </inertial>
  </link>
  <gazebo reference="left_wheel_base_joint">
    <material>Gazebo/Blue</material>
  </gazebo>
  <joint name="left_wheel_base_joint" type="continuous">
    <parent link="chassis"/>
    <child link="left_wheel_base"/>
    <origin rpy="0.0 0.0 0.0" xyz="-0.08 0 -0.0075"/>
  </joint>
  <!-- Left_wheel-->
  <link name="left_wheel">
    <visual>
      <geometry>
        <cylinder length="0.009" radius="0.033"/>
      </geometry>
      <origin rpy="0 0.0 1.5707963267948966" xyz="0 0 0.0"/>
      <material name="grey"/>
    </visual>
  </link>
  <joint name="left_wheel_joint" type="continuous">
    <parent link="left_wheel_base"/>
    <child link="left_wheel"/>
    <origin xyz="0 0 -0.159"/>
    <axis xyz="0 1 0"/>
  </joint>
  <!-- right_wheel_base-->
  <link name="right_wheel_base">
    <visual>
      <geometry>
        <box size="0.015 0.015 0.05"/>
      </geometry>
      <origin xyz="0.08 0 -0.0325"/>
      <material name="white"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.017 0.015 0.015"/>
      </geometry>
      <origin xyz="0.08 0 -0.0575"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.015 0.015 0.076"/>
      </geometry>
      <origin xyz="0.10450000000000001 -0.038 0.0"/>
    </visual>
    <visual>
      <geometry>
        <box size="0.0115 0.015 0.015"/>
      </geometry>
      <origin xyz="0.00575 0 -0.14100000000000001"/>
    </visual>
    <inertial>
      <mass value="1"/>
      <inertia ixx="1e-3" ixy="0.0" ixz="0.0" iyy="1e-3" iyz="0.0" izz="1e-3"/>
    </inertial>
  </link>
  <gazebo reference="right_wheel_base_joint">
    <material>Gazebo/Blue</material>
  </gazebo>
  <joint name="right_wheel_base_joint" type="continuous">
    <parent link="chassis"/>
    <child link="right_wheel_base"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.08 0 -0.0075"/>
  </joint>
  <!-- right_wheel-->
  <link name="right_wheel">
    <visual>
      <geometry>
        <cylinder length="0.009" radius="0.033"/>
      </geometry>
      <origin rpy="0 0.0 1.5707963267948966" xyz="0 0 0.0"/>
      <material name="grey"/>
    </visual>
  </link>
  <gazebo reference="right_wheel">
    <material>Gazebo/Black</material>
  </gazebo>
  <joint name="right_wheel_joint" type="continuous">
    <parent link="right_wheel_base"/>
    <child link="right_wheel"/>
    <origin xyz="0 0 -0.159"/>
    <axis xyz="0 1 0"/>
  </joint>
  <!-- castor wheel-->
  <link name="castor_wheel">
    <visual>
      <geometry>
        <box size="0.015 0.015 0.14100000000000001"/>
      </geometry>
      <material name="white"/>
    </visual>
    <visual>
      <geometry>
        <sphere radius="0.033"/>
      </geometry>
      <origin xyz="0 0 -0.159"/>
    </visual>
  </link>
  <joint name="castor_joint" type="fixed">
    <parent link="chassis"/>
    <child link="castor_wheel"/>
    <origin xyz="0 0.2 -0.159"/>
  </joint>
</robot>
